catroid/src/main/java/org/catrobat/catroid/utils/CrashReporter.java

```diff
@@ -94,17 +88,9 @@ public static boolean logException(Throwable t) {
 	}
 
 	public static boolean checkIfCrashRecoveryAndFinishActivity(final Activity context) {
-		Log.d(TAG, "AFTER_EXCEPTION : checkIfCrashRecoveryAndFinishActivity()");
-		if (isRecoveredFromCrash()) {
-			if (isReportingEnabled()) {
-				sendUnhandledCaughtException();
-			}
-			if (!(context instanceof MainMenuActivity)) {
-				context.finish();
-			} else {
-				preferences.edit().putBoolean(RECOVERED_FROM_CRASH, false).commit();
-				return true;
-			}
+		if (isReportingEnabled()) {
+			sendUnhandledCaughtException();
+			return true;
 		}
 		return false;
 	}
```

Comment:
What I have understood is that checkIfCrashRecoveryAndFinishActivity() method should be removed and its features (which is required) should be implemented in sendUnhandledCaughtException(). So that's what I have to do?